kmk/hid.py

```diff
@@ -221,3 +222,109 @@ def hid_send(self, evt):
         return self.devices[reporting_device_const].send_report(
             evt[1 : HID_REPORT_SIZES[reporting_device_const] + 1]
         )
+
+
+class BLEHID(AbstractHID):
+    try:
+        # TODO FIXME This is wrapped in a try block to prevent crashes on M4 devices
+        from adafruit_ble import BLERadio
+        from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
+        from adafruit_ble.services.standard.hid import HIDService
```

Comment:
shouldn't these be inside `__init__`?

Reply:
If it didn't blow memory out on devices that have these libs, but aren't using BLE, yes. I need to get all of this section cleaned though. Thanks for the reminder.

Reply:
oh you mean just the class definition blows up the memory? damn. Maybe we can split all the HID implementations in different files and import them at will

Reply:
adafruit_ble is pretty heavy alone. The main reason it's in the try block is boards like the M0 (Itsy bitsy) will crash without the try on it. This needs to at least be split into BLE/USB.

Reply:
I don't like the idea of wrapping the world in a giant try/catch. Move these imports to the top of the file and wrap the imports alone in a try/catch if necessary for whatever reason. We can solve the memory blowout in docs - that's the whole point of extensions. If you don't need `BLEHID`, don't' import it or you'll explode your RAM. If you do need it, run KMKPython. Seems straightforward to me unless I'm missing / forgetting some context here.

Reply:
That's fair enough then.